File: my_linalg_lib/my_linalg_lib.jl
using Base

function my_matmul(A, B)
    m, n = size(A)
    n, p = size(B)
    C = Array{Float64}(undef, m, p)
    for i in 1:m
        for j in 1:p
            C[i, j] = 0
            for k in 1:n
                C[i, j] += A[i, k]*B[k, j]
            end
        end
    end
    return C
end

A = [1 2
     3 4]

B = [-1 0
     0 -1]

C_expected = [-1 -2
              -3 -4]

@assert A * B == C_expected
@assert my_matmul(A, B) == C_expected

File: my_linalg_lib/my_linalg_lib.py
def matmul(A, B):
    C = []
    
    for i in range(len(A)):
        C.append([None]*len(A[i]))
        for j in range(len(B)):
            C[i][j] = 0
            for k in range(len(A[i])):
                C[i][j] += A[i][k] * B[k][j]
    return C

A = [[1, 2], [3, 4]]
B = [[1, 0], [0, 1]]

C_expected = [[1, 2], [3, 4]]

assert matmul(A, B) == C_expected
